### ISSUE TITLE
3D Items Render Issue connected to bow changes? 

### ISSUE THREAD

**Issue by LOR-Studio** [jrbudda/minecrift#256]
Hi, 
  
  First off, thank you for all the hard work you have done on this project. You really have made something special here, and I am surprised it hasn't caught on more. To my question though, Is there any way to revert back to a older revision before you added the bow update that made it where you had to pull back the bows to shoot? Now, I like all the other changes and fixes you have made, but I believe the bow changes conflicts with mods. I've have tried installing older versions, particularly v13 the one before the bow update, but it crashes, even if I do it completely fresh install with no mods. Another thing I've noticed is that some older versions wont finish downloading from the repo and will never launch. The new version works fine no issues except...Gun mods, some work better then others but the main thing I've noticed is the way it renders in your hand completely wrong position and some mod weapons will not shoot. Ill give you an example, Balkon's WeaponMod some of the guns on there you have to pretend they are a bow to shoot them, it puts a arrow in your hand with the blunderbuss. 
   
  In your older Revs, I used Stefinus 3D guns mod, an they rendered mostly in a correct position in your hand, but issue I had back then was that it seemed to be unstable and crash with certain mods. As of now, I'm trying to use the TechGuns mod, issues are: they don't position right in hand, and wont shoot even if I rebind to a keyboard button. Npcs that spawn with guns are holding them correctly. I really don't want to revert back to a older rev. I just feel like being able to hold and shoot so guns would add a lot of fun scenarios you could do with friends. If there is any way fix it, without reverting back would be awesome! Thanks!

**Comment by Techjar** [comment]
Yeah we can't possibly make every single mod item render in the correct orientation, without writing a custom orientation for every mod, so some are going to look weird. As for the keyboard binding issue, I don't know, but I have seen mods use the key binds in a stupid way that makes them not work.

**Comment by jrbudda** [comment]
Our primary focus is on gameplay, not sure why the bow update would break a mod, tho a lot relies on exactly how the mod author codes their weapons. The 2 handed archery should be limited to ItemBow but maybes theres something that interferes with its inheritors. An option to disable the 2 handed archery isnt a terrible idea if that's the case. I make no promises on the cosmetics as techjar said thats a major pain.

**Comment by LOR-Studio** [comment]
That would be great, just to be able to disable the two handed bow mechanic. I could live without it myself. I found another mod though, I tested last night on the latest rev. Its http://rivalrebels.blogspot.com/, The weapons will shoot on using secondary attack or default press down on right touch pad. It works surprising well, except the weapons are bent down facing the ground instead of facing forward. I understand how complicated changing the code would be for every mod, I don't know anything really about Java my self. I decompiled some packages and I honestly wouldn't want to touch it with a ten foot pool. I appreciate the reply's though, Thanks!

**Comment by fucnknown** [comment]
Hi, I want to bring back an idea from when the bow mechanic ( #213) was implemented and brought the item rendering orientation issue.

Let's assume you create a method for tweaking orientation of a given object by its ID.
Definition could be stored as:
**ID: X Y Z I J K F**

If the item is not defined, use standard rendering.
If the item is defined then "tweak" its rendering:
-  X degrees on the X axis
-  Y Degrees in the Y axis
-  Z degrees in the Z axis,
- I Pixels forward from starting position in the X axis
- J pixels along the Y axis
- K pixels along the Z axis
- F would be the point of origin of projectiles.-
Other variables that you adjust when manually could be included, for example a boolean to define "Bow behavior" or things like that...

Then you leave the community the job of finding the correct values for each item they want to fix. So those who play TiC and want the crossbow working can fiddle the values until it's working.

**Comment by Techjar** [comment]
I've thought about doing such a thing, but there's a few issues. The way Euler angles get handled for rendering isn't very intuitive, so we'd need to somehow fix that. I'm also not sure if anyone would really care enough to use this feature, since tweaking the values would be quite tedious.

**Comment by fucnknown** [comment]
I can't really judge the Euler's transformations complexity, but leaving the raw values exposed (while either sanity checking the extremes and/or inform them)  should be enough. 
As far as the tedioussness of fixing an item you might be underestimating your user base;  these are Minecraft nerds,  modded Minecraft nerds at that,  most are used to tweaking values and configurations to make them work. If the values can be tweaked in the fly (while playing), That would be swell, though.   Finally if we are using named id's, particular mod configurations can be shared among users reducing the total effort required...

**Comment by Chimera245** [comment]
I second this! I am one such Minecraft Nerd(tm) who would go through such tedium to get my favorite items working in VR. 
And then once any one user found a pleasant orientation for any one item, he could upload the post-crunched numbers to any popular modding message board, and users could start a repository of item orientation tweaks.
